.github/workflows/ci.yml

```diff
@@ -8,44 +8,74 @@ env:
 
 jobs:
   checkers:
+    name: Run static checkers
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
     - name: Run clang-format style check (.c and .h)
       uses: jidicula/clang-format-action@master
 
-  build:
+  sanitizers:
+    name: Test with -fsanitizer=${{ matrix.sanitizer }}
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        sanitizer: [thread, undefined] # LEAK TO BE CORRECTED: leak, address
+    steps:
+    - name: Prepare
+      run: sudo apt install libevent-dev
+    - uses: actions/checkout@v2
+    - name: Create build folder
+      run: cmake -E make_directory build
```

Comment:
This looks very similar to the regular build and tests. Duplicated Yaml code?

Do we run tests both with and without sanitizers? Can't we run the CI build & tests with santizers always on? Only performance tests would make sense to run without such tools IMO, or what do you think?

Reply:
The regular build have a build matrix with gcc and clang to find issues like initiation order problem and other stuff, rather annoying:
https://stackoverflow.com/questions/25669493/c-why-does-llvm-evaluate-printf-left-to-right-when-gcc-evaluates-right-to-left
And this job tests for leaks. Maybe there are ways to optimize this.
 I will move this to another PR with sanitizer and leakfix